src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Source/SourceUpdaterStep.cs

```diff
@@ -50,16 +51,22 @@ public class SourceUpdaterStep : UpgradeStep
             WellKnownStepIds.NextProjectStepId,
         };
 
-        public SourceUpdaterStep(IEnumerable<DiagnosticAnalyzer> analyzers, IEnumerable<CodeFixProvider> codeFixProviders, ILogger<SourceUpdaterStep> logger)
+        public SourceUpdaterStep(IEnumerable<DiagnosticAnalyzer> analyzers, IEnumerable<CodeFixProvider> codeFixProviders, IEnumerable<AdditionalText> additionalTexts, ILogger<SourceUpdaterStep> logger)
```

Comment:
Again, why not pass an ImmutableArray<AdditionalText>?

Reply:
Same reason (ctor injection typically uses IEnumerable). I'll check to see if ImmutableArray works here and switch the code if it does.